dagfactory/dagbuilder.py

```diff
@@ -369,77 +379,103 @@ def make_task(operator: str, task_params: Dict[str, Any]) -> BaseOperator:
-                            else None
-                        )
-                    task_params["full_pod_spec"] = (
-                        V1Pod(**task_params.get("full_pod_spec"))
-                        if task_params.get("full_pod_spec") is not None
-                        else None
-                    )
-                    task_params["init_containers"] = (
-                        [V1Container(**v) for v in task_params.get("init_containers")]
-                        if task_params.get("init_containers") is not None
+                task_params["volume_mounts"] = (
+                    [VolumeMount(**v) for v in task_params.get("volume_mounts")]
+                    if task_params.get("volume_mounts") is not None
+                    else None
+                )
+
+                task_params["volumes"] = (
+                    [DagBuilder._create_volume(vol) for vol in task_params["volumes"]]
+                    if task_params.get("volumes") is not None
+                    else None
+                )
+
+                task_params["env_vars"] = (
+                    [V1EnvVar(**env_var) for env_var in task_params.get("env_vars")]
+                    if task_params.get("env_vars") is not None
+                    else None
+                )
+
+                task_params["env_from"] = (
+                    [V1EnvFromSource(**v) for v in task_params["env_from"]]
+                    if task_params.get("env_from") is not None
+                    else None
+                )
+
+                task_params["secrets"] = (
+                    [Secret(**v) for v in task_params.get("secrets")]
+                    if task_params.get("secrets") is not None
+                    else None
+                )
+
+                task_params["container_resources"] = (
+                    V1ResourceRequirements(**task_params["container_resources"])
+                    if task_params.get("container_resources") is not None
+                    else None
+                )
+
+                task_params["affinity"] = (
+                    V1Affinity(task_params["affinity"]) if task_params.get("affinity") is not None else None
+                )
+
+                task_params["image_pull_secrets"] = (
+                    [V1LocalObjectReference(**v) for v in task_params["image_pull_secrets"]]
+                    if task_params.get("image_pull_secrets") is not None
+                    else None
+                )
+
+                if version.parse(K8S_PROVIDER_VERSION) >= version.parse("7.8.0"):
+                    # See PR: https://github.com/apache/airflow/pull/35063
+                    task_params["host_aliases"] = (
+                        [V1HostAlias(**v) for v in task_params.get("host_aliases")]
+                        if task_params.get("host_aliases") is not None
                         else None
                     )
 
+                task_params["tolerations"] = (
```

Comment:
This method is quite complex, and we repeatedly apply the same logic for each parameter.

Would this be an opportunity to extract the logic for the K8S pod operator in a separate function and try to reuse a bit of the code that seems duplicated? Do we need to set as `None` values that are not defined in the YAML?

Some thoughts on a potentially refactored version (not tested):

```
key_to_class = {
    "tolerations": V1Toleration,
    "security_context": V1PodSecurityContext,
    ...
}

for param_key, param_values in task_params.items():
    task_params[param_key] = key_to_class[param_key](param_values)
```

Reply:
I have tried to clean it a bit in commit: 07387cabf267cc786da49da0ba40792071801591